### PR TITLE
fix(channel): do not resuse existing channel IDs

### PR DESCRIPTION
**Pull request by mpastyl**
Check if channel ID is in use when creating a new channel. This might happen if channel ids have warped around their max value.

### CODE REVIEW

File: core/channel.go [hunk 1]
```diff
@@ -112,23 +112,32 @@ type Channel struct {
...
+		// store API channel within the client
+		c.channelsLock.Lock()
+		_, found := c.channels[chID]
+		if !found {
+			c.channels[chID] = channel
+			c.channelsLock.Unlock()
+			break
+		}
+		c.channelsLock.Unlock()
+	}
```
Comment (mejedi) [inline core/channel.go:140]: We throw away a Channel if the id is already taken, which is a bit wasteful.

Maybe we could create a channel first, without setting the id. Then grab the mutex and repeatedly set the id to the next one and break once we got a unique one?

There's no problem in bumping the counter while holding the lock; it's dirt cheap.

Comment (mpastyl) [reply core/channel.go:133]: Thanks, that makes sense. I assume we no longer need the atomic in this case (maxChannelD is not used elsewhere).

**Review by mejedi — APPROVED** [review]
Please commit squashed.

**Comment by mpastyl** [comment]
Check if channel ID is in use when creating a new channel.
This might happen if channel IDs have warped around their max value.

**Pull request merged**